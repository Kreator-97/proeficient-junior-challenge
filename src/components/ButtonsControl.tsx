import { useAppDispatch } from '../app/hooks'
import { reset } from '../app/slices/summarySlice'

export const ButtonsControl = () => {

  const dispatch = useAppDispatch()

  return (
    <div className="flex justify-between mb-12 px-4">
      <button
        onClick={ () => dispatch( reset() ) }
        className="btn bg-white rounded-sm border-2 border-primary hover:bg-primary"
      >
        Clear
      </button>
    </div>
  )
}
