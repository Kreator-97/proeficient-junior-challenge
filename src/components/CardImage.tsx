import { FC } from 'react'

interface Props {
  imgURL  : string;
  title   : string;
}

export const CardImage: FC<Props> = ({imgURL, title}) => {
  return (
    <div className='h-16 flex items-center'>
      <img
        className='max-h-[48px]'
        src={imgURL}
        alt={title}
        width={48}
      />
    </div>
  )
}
