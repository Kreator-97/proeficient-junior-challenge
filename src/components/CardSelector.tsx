import { FC } from 'react'
import { Product } from '../interfaces'
import { CardImage, Selector } from './'

interface Props {
  product: Product;
}

export const CardSelector: FC<Props> = ({ product }) => {

  return (
    <div>
      <div className="flex flex-col gap-4 justify-center p-4 items-center mb-4 shadow-lg border border-gray-100">
        <CardImage
          imgURL={ product.imgURL }
          title={ product.title }
        />
        <div>
          <p className='block w-full'>{product.title}</p>

        </div>
      </div>
      <Selector />
    </div>
  )
}
