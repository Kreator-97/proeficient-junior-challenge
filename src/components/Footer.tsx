export const Footer = () => {
  return (
    <footer className="min-h-[200px] bg-primary"></footer>
  )
}
