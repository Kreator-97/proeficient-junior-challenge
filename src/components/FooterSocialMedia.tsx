import { IoLogoFacebook, IoLogoTwitter } from 'react-icons/io'
import { AiFillInstagram } from 'react-icons/ai'

export const FooterSocialMedia = () => {
  return (
    <div className='flex gap-4 mx-auto md:mx-0'>
      <div
        style={{borderWidth: '3px' }}
        className='border-white p-2 rounded-full'>
        <IoLogoTwitter size={24} color='white'/>
      </div>

      <div
        style={{borderWidth: '3px' }}
        className='border-white p-2 rounded-full'
      >
        <IoLogoFacebook size={24} color='white'/>
      </div>

      <div
        style={{borderWidth: '3px' }}
        className='border-white p-2 rounded-full'
      >
        <AiFillInstagram size={24} color='white'/>
      </div>
    </div>
  )
}
