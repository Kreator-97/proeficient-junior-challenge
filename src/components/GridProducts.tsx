import { FC } from 'react'
import { Product } from '../interfaces'
import { CardSelector } from '.'

interface Props {
  products: Product[]
}

export const GridProducts: FC<Props> = ({products}) => {
  return (
    <div className='grid gap-8 grid-cols-1 sm:grid-cols-2  md:grid-cols-4 lg:grid-cols-5 px-4 mb-8'>
      {
        products.map((product) => {
          console.log(product)
          return (
            <CardSelector key={product.title} product={product} />
          )
        })
      }
    </div>
  )
}
