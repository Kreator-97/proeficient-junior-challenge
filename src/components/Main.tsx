import { ButtonsControl, GridProducts, Summary } from './'
import { products } from '../data/products'

export const Main = () => {
  return (
    <div className='max-w-5xl mx-auto '>
      <section className="text-center my-20">
        <h1 className="text-5xl text-dark py-8">What items to store?</h1>
        <p>Select which items you wish to store before moving to your new home. We&apos;ll keep &apos;em safe</p>

      </section>

      <GridProducts products={products} />
      <ButtonsControl />
      <Summary />
    </div>
  )
}
