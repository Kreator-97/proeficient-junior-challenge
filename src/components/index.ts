export * from './ButtonsControl'
export * from './CardImage'
export * from './CardSelector'
export * from './GridProducts'
export * from './Main'
export * from './Navbar'
export * from './Footer'
export * from './Selector'
