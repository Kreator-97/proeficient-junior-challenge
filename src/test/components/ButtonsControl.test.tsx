import { screen, render } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import { Provider } from 'react-redux'

import { ButtonsControl } from '../../components/ButtonsControl'
import { store } from '../../app/store'

const dispatch = vi.fn()

vi.mock('../../app/hooks', () => {
  return {
    useAppDispatch: () => dispatch
  }
})

describe('test on ButtonControl component', () => {
  it('should to match with snapshot', () => {

    const { container  } = render(
      <Provider store={store}>
        <ButtonsControl />
      </Provider>
    )

    expect( container ).toMatchSnapshot()
  })

  it('dispatch should to dispatch the reset action', () => {
    
    const resetAction = { payload: undefined, type: 'summary/reset' }
    
    render(
      <Provider store={store}>
        <ButtonsControl />
      </Provider>
    )

    screen.getByRole('button').click()

    expect( dispatch ).toHaveBeenCalledWith(resetAction)
  })
})
