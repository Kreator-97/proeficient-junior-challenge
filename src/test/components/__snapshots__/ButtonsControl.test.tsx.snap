// Vitest Snapshot v1

exports[`test on ButtonControl component > should to match with snapshot 1`] = `
<div>
  <div
    class="flex justify-between mb-12 px-4"
  >
    <button
      class="btn bg-white rounded-sm border-2 border-primary hover:bg-primary"
    >
      Clear
    </button>
  </div>
</div>
`;
